src/test/scala/io/iohk/ethereum/db/storage/TransactionalKeyValueStorageSuite.scala

```diff
@@ -0,0 +1,122 @@
+package io.iohk.ethereum.db.storage
+
+import io.iohk.ethereum.ObjectGenerators
+import io.iohk.ethereum.db.dataSource.{DataSource, EphemDataSource}
+import io.iohk.ethereum.rlp.RLPImplicits._
+import io.iohk.ethereum.rlp.{decode => rlpDecode, encode => rlpEncode}
+import org.scalacheck.Gen
+import org.scalatest.FunSuite
+import org.scalatest.prop.PropertyChecks
+
+class TransactionalKeyValueStorageSuite extends FunSuite with PropertyChecks with ObjectGenerators {
+  val iterationsNumber = 100
+
+  object IntStorage {
+    val intNamespace: IndexedSeq[Byte] = IndexedSeq[Byte]('i'.toByte)
+    val intSerializer: Int => IndexedSeq[Byte] = (i: Int) => rlpEncode(i).toIndexedSeq
+    val intDeserializer: IndexedSeq[Byte] => Int =
+      (encodedInt: IndexedSeq[Byte]) => rlpDecode[Int](encodedInt.toArray)
+  }
+
+  class IntStorage(val dataSource: DataSource) extends TransactionalKeyValueStorage[Int, Int] {
+    import IntStorage._
+
+    override val namespace: IndexedSeq[Byte] = intNamespace
+    override def keySerializer: Int => IndexedSeq[Byte] = intSerializer
+    override def valueSerializer: Int => IndexedSeq[Byte] = intSerializer
+    override def valueDeserializer: IndexedSeq[Byte] => Int = intDeserializer
+  }
+
+  def newIntStorage(): IntStorage = new IntStorage(EphemDataSource())
+
+  test("Get ints from KeyValueStorage") {
+    forAll(Gen.listOf(intGen), Gen.listOf(intGen)) { (intsInStorage, unfilteredIntsNotInStorage) =>
+      val intsNotInStorage = unfilteredIntsNotInStorage.distinct diff intsInStorage
```

Comment:
Given how this is generated, couldn't it often be the case that intsNotInStorage is empty?